### PR TITLE
Amend `scanner.c` to correctly parse strings of the form `"*\$"`

### PR DESCRIPTION
**Pull request by yosefAlsuhaibani**
For some reason, the `scan_string_content` cannot parse a string that has an escaped dollar sign at the end of the string, like `abc\$` or `\$`; I've included a very small fix to `scanner.c` that solves this.

### CODE REVIEW

File: test/corpus/literals.txt [hunk 1]
```diff
@@ -114,7 +114,7 @@ More string interpolation
 "$1"
 "$ $ $"
 "\$foo"
-
+"\$"
```
Comment (fwcd) [inline test/corpus/literals.txt:117]: Style nit:

```suggestion
"\$"

```

**Review by fwcd — COMMENTED** [review]
Thanks for spotting this! Just a minor nit, otherwise this looks good.

File: test/corpus/literals.txt (hunk 1 above)
Comment (yosefAlsuhaibani) [reply test/corpus/literals.txt:117]: Oops! thank you for spotting this.

**Comment by yosefAlsuhaibani** [comment]
Of course: thank you for maintaining this parser!

**Pull request merged**